flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/file/lsm/FileStoreImpl.java

```diff
@@ -100,23 +107,29 @@ public FileStoreImpl(
 
     @Override
     public void put(RowData key, RowData value) throws StoreException {
-        memTable.put(levels.newSequenceNumber(), ValueKind.ADD, key, value);
-        checkFlush();
+        try {
+            memTable.put(levels.newSequenceNumber(), ValueKind.ADD, key, value);
+        } catch (StoreException e) {
+            doFlush();
+            put(key, value);
+        }
     }
 
     @Override
     public void delete(RowData key, RowData value) throws StoreException {
-        memTable.put(levels.newSequenceNumber(), ValueKind.DELETE, key, value);
-        checkFlush();
+        try {
+            memTable.put(levels.newSequenceNumber(), ValueKind.DELETE, key, value);
+        } catch (StoreException e) {
+            doFlush();
+            delete(key, value);
```

Comment:
`memTable.put(levels.newSequenceNumber(), ValueKind.DELETE, key, value);` and just throws exception ("the write buffer is too small") if fail.